### PR TITLE
Kustomize, Prometheus, and Merak Agent Refactor

### PR DESCRIPTION
**Pull request by phudtran**
This PR adds the following

- Refactor Merak Agent EVM operations into a new EVM abstraction
- Refactor Merak Agent Standalone as an environment variable toggle
- Instrument EVM operations with following prometheus metrics
- Created a basic metrics abstraction that collects the following metrics
  - Operation Latency -  HistogramVec
  - Operation Success - CounterVec
  - Operation Failure -    CounterVec
- Created Prometheus config to scrape vhost pods
- Created Prometheus Kubernetes deployment yaml
- Refactor Merak deployment with Kustomize
- Adds Kustomize files for the following scenarios
  - CI
  - compute.test
  - dev ( Includes Alcor)

To start a prometheus server on your component you can do the following
```
	go func() {
		hostname, err := os.Hostname()
		if err != nil {
			log.Fatal("Unable to get hostname!")
		}
		PrometheusRegistry := prometheus.NewRegistry()
		MerakMetrics := metrics.NewMetrics(handler.PrometheusRegistry, "ComponentName"+hostname)
		http.Handle("/metrics", promhttp.HandlerFor(
			handler.PrometheusRegistry,
			promhttp.HandlerOpts{Registry: handler.PrometheusRegistry}))
		http.ListenAndServe(fmt.Sprintf(":%d", *gRPCPort), nil)
	}()
```

This will create a new metrics struct with name "ComponentName_hostname" and start a new prometheus server in a separate thread.

To instrument your function and get latency numbers you can then add these two lines at the beginning of your functions
```	
var err error
defer metrics.GetMetrics(MerakMetrics, &err)()
```

Finally, you will need to modify the file `config/prometheus.yaml` and add your pod to be scraped. Currently only vhosts pods will be scraped.
Once modified, rebuild and push the prometheus image using the file `docker/prometheus.Dockerfile`

### CODE REVIEW

File: deployments/kubernetes/base/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+# Kustomize Base Files
```
Comment (cj-chung) [inline deployments/kubernetes/base/README.md:1]: @phudtran can you add some sentences to describe what this folder contains and what are they use for?

File: deployments/kubernetes/ci/README.MD [hunk 2]
```diff
@@ -0,0 +1 @@
+# Kustomize CI Overlay
```
Comment (cj-chung) [inline deployments/kubernetes/ci/README.MD:1]: @phudtran add some sentence to describe what this folder contains too.

File: deployments/kubernetes/compute.test/README.MD [hunk 3]
```diff
@@ -0,0 +1 @@
+# Kustomize Merak-Compute test overlay
```
Comment (cj-chung) [inline deployments/kubernetes/compute.test/README.MD:1]: @phudtran  same here.

File: deployments/kubernetes/dev/README.MD [hunk 4]
```diff
@@ -0,0 +1 @@
+# Kustomize Merak-Compute test develeopment
```
Comment (cj-chung) [inline deployments/kubernetes/dev/README.MD:1]: @phudtran same here, add some sentence to describe that this folder contains and what are they use for.

**Review by cj-chung — COMMENTED** [review]
@phudtran For the metrics wrapper, it's better have a simple example to show how to use it.

File: deployments/kubernetes/ci/README.MD (hunk 2 above)
Comment (phudtran) [reply deployments/kubernetes/ci/README.MD:1]: Ok, I've updated the README

File: deployments/kubernetes/compute.test/README.MD (hunk 3 above)
Comment (phudtran) [reply deployments/kubernetes/compute.test/README.MD:1]: Ok, I've updated the README

File: deployments/kubernetes/dev/README.MD (hunk 4 above)
Comment (phudtran) [reply deployments/kubernetes/dev/README.MD:1]: Ok, I've updated the README

**Comment by phudtran** [comment]
> @phudtran For the metrics wrapper, it's better have a simple example to show how to use it.

I have added instructions in the PR on how to use prometheus and the metrics wrapper.

**Review by cj-chung — APPROVED** [review]
LGTM

**Pull request merged**